app/models/transfer_package.rb

```diff
@@ -29,6 +29,19 @@ class TransferPackage < ApplicationRecord
           where(receiver_institution_id: institution.id)
         }
 
+  before_create do
+    sample_transfers.each do |sample_transfer|
+      sample = sample_transfer.sample
+      unless confirmed?
+        sample.detach_from_context
+      end
+      if includes_qc_info
+        sample.attach_qc_info
+      end
+      sample.save!
+    end
```

Comment:
Maybe use some oneliners to reduce indentation?

```ruby
sample = sample_transfer.sample
sample.detach_from_context unless confirmed?
sample.attach_qc_info if includes_qc_info
sample.save!
```

Feel free to overlook, it's pedantic style.